### PR TITLE
C++

### PR DESCRIPTION
**Pull request by DiegoAlbertoTorres**
A C++ plugin for Beaker!
Closes https://github.com/twosigma/beaker-notebook/issues/1908

### CODE REVIEW

File: plugin/cpp/src/dist/cpp.js [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+      }
+    },
+    resetEnvironment: function() {
+      $.ajax({
+        type: 'POST',
+        datatype: 'json',
+        url: bkHelper.serverUrl(serviceBase + '/rest/cpp/resetEnvironment'),
+        data: {shellId: this.settings.shellID}
+      }).done(function(ret) {
+        console.log('done resetEnvironment', ret);
```
Comment (samccone) [inline plugin/cpp/src/dist/cpp.js]: do we want all of these logs?

Comment (DiegoAlbertoTorres) [inline plugin/cpp/src/dist/cpp.js]: Huh. So all that stuff is a clone of the javash plugin, so I do not know. @scottdraves do we want these logs?

**Comment by scottdraves** [comment]
```

:dev:plugin:cpp:makeCRunLibIn file included from CRun.cpp:16:
./CppKernel.h:2:10: fatal error: 'jni.h' file not found
#include <jni.h>
         ^
1 error generated.
 FAILED
```

File: plugin/cpp/src/dist/cpp.js (hunk 1 above)
Comment (scottdraves) [inline plugin/cpp/src/dist/cpp.js]: not really

**Comment by samccone** [comment]
do we need to update our linux deps? / Dockerfile?

Do we want to add an eval test to travis?

**Comment by DiegoAlbertoTorres** [comment]
I don't think so. I am yet to try to run this on Linux.
Also, yes we probably want this on Travis.

**Comment by scottdraves** [comment]
```
:dev:plugin:cpp:makeCRunLibIn file included from CRun.cpp:16:
In file included from ./CppKernel.h:2:
/usr/lib/jvm/java-7-oracle//include/jni.h:45:10: fatal error: 'jni_md.h' file not found
#include "jni_md.h"
         ^
1 error generated.
 FAILED
```

**Pull request merged**